### PR TITLE
Clic pflow merge

### PR DESCRIPTION
**Pull request by sunnyw22**
Trainings been improved

Modifications made:
- Env: Added jupyter notebook and ipykernel (not necessary but added for the notebooks), added gxx_linux-64 (the "CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage" is driving me crazy, added to solve this, but I know we don't see this error on the other branches, might be something wrong with my env setup even I have recreated the env)
- Trainings: Allow multiple GPUs training and slight modification on the intermediate losses
- Losses: We now have two versions of the losses, one averages the loss by the total number of elements (main), while the other averages the loss by the batch size (dmitrii) (will remove the averaging by batch size soon)
- CLI: `after_instantiate_classes` did not work for me somehow (env related again, can be reverted)

Problems before: 
- Unstable training loss curve (fixed by updating the loss and cost functions, ensuring costs computed with f32)
Problems now:
- Stable training curves with good convergence, but poor validation on object classification (seems to be fixed by allowing intermediate loss, needed to wait trainings to finish to confirm)

### CODE REVIEW

File: src/hepattn/models/task.py [hunk 1]
```diff
@@ -443,6 +431,7 @@ def __init__(
         loss_class_weights: list[float] | None = None,
         null_weight: float = 1.0,
         mask_queries: bool = False,
+        inter_loss: bool = True,
```
Comment (samvanstroud) [inline src/hepattn/models/task.py:434]:
```suggestion
        has_intermediate_loss: bool = True,
```
for consistency

File: src/hepattn/models/task.py [hunk 2]
```diff
@@ -477,6 +466,7 @@ def __init__(
         self.losses = losses
         self.costs = costs
         self.num_classes = num_classes
+        self.has_intermediate_loss = inter_loss
```
Comment (samvanstroud) [inline src/hepattn/models/task.py:469]: Does it make sense to have this be configurable for all tasks? i.e. have it as an argument to the task base class nad pass it via `super().__init__()`?

**Review by samvanstroud — COMMENTED** [review]
Thanks @sunnyw22! I'm going to merge this and maybe we can address some more comments in the target branch before it hits main
